core/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -56,6 +57,25 @@ public int getNumberOfPendingTasks() {
         return size;
     }
 
+    /**
+     * Returns the waiting time of the first task in the queue
+     */
+    public TimeValue getMaxTaskWaitTime() {
+        if (getQueue().size() == 0) {
+            return NO_WAIT_TIME_VALUE;
+        }
+
+        long oldestCreationDateInNanos = System.nanoTime();
+        for (Runnable queuedRunnable : getQueue()) {
+            if (queuedRunnable instanceof PrioritizedRunnable) {
+                oldestCreationDateInNanos = Math.min(oldestCreationDateInNanos,
+                        ((PrioritizedRunnable) queuedRunnable).getCreationDateInNanos());
+            }
+        }
+
+        return TimeValue.timeValueNanos(System.nanoTime() - oldestCreationDateInNanos);
```

Comment:
Nit picky - Can we capture the initial ` long oldestCreationDateInNanos = System.nanoTime();` and use this as the "now"? Just worried that the queue can be empty (after the initial check) and we will still get a non 0 value ..